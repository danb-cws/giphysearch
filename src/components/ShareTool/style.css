.ShareTool--open-button {
  font-size: 1rem;
  background-image: var(--share-icon-b64-url);
  background-size: cover;
  background-color: transparent;
  width: 2em;
  height: 2em;
  border: none;
  -webkit-appearance: none;
  cursor: pointer;
  position: absolute;
  right: var(--primary-horizontal-gutter);
  bottom: var(--primary-vertical-gutter);
  z-index: 1;
}
.ShareTool--panel {
  position: absolute;
  top: 2rem;
  left: 2rem;
  right: 2em;
  bottom: 2em;
  padding: 2em;
  background: black;
  color: white;
}

.modal {
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
}
.modal:before {
  content: "";
  position: fixed;
  background-color: rgba(0, 4, 0, 0.8);
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
}
.modal:after {
  content: "x";
  width: var(--modal-close-size);
  height: var(--modal-close-size);
  position: absolute;
  z-index: 3;
  top: calc(50% - (var(--modal-height) / 2) + var(--modal-close-size));
  right: calc(50% - (var(--modal-width) / 2) - var(--modal-close-size));
  color: white;
  background-color: blue;
  cursor: pointer;
}
.modal > * {
  z-index: 2;
  position: relative;
  width: var(--modal-width);
  height: var(--modal-height);
}
/*
	.any-custom-class {
		background: lavenderblush;
		padding: 1rem;
	}

	</style>
</head>

<body>



<div class="modal" onclick="event.target.classList.contains('modal') ? this.parentNode.removeChild(this) : null;">
	<div class="any-custom-class">
		<div>Some content</div>
		<div>Maybe a <a href="http://www.google.com?q=cats">link</a>?</div>
	</div>
</div>

<p>
	Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum. 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod 	tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, 	quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo 	consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse 	cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non 	proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
</p>

</body>
</html>*/
