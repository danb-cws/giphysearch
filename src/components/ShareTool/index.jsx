import React, { Component } from "react";
import "./style.css";

class ShareTool extends Component {
  state = {
    isShowing: false
  };

  toggleTool = () => {
    console.log("togglr");
    this.setState({ isShowing: true });
  };

  render() {
    return !this.state.isShowing ? (
      <button className="ShareTool" onClick={this.toggleTool}>
        sharetool
      </button>
    ) : (
      <div className="ShareTool">the tool panel</div>
    );
  }
}
export default ShareTool;
