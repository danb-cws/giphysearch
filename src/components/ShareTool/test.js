import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import ShareTool from "./index";
import testjson from "mock_json/gifMock.json";

describe("ShareTool tests", () => {
  it("Renders the ShareTool component", () => {
    const el = shallow(<ShareTool shareUrl={testjson.embed_url} />);
    expect(el.exists()).toBe(true);
  });

  it("Renders and matches snapshot (closed state)", () => {
    const tree = shallow(<ShareTool shareUrl={testjson.embed_url} />);
    expect(tree).toMatchSnapshot();
  });

  it("Renders and matches snapshot (open state)", () => {
    global.document.queryCommandSupported = () => {
      return {
        copy: () => {
          true;
        }
      };
    };
    const tree = mount(<ShareTool shareUrl={testjson.embed_url} />);
    tree.find("button.ShareTool--open-button").simulate("click");
    expect(tree).toMatchSnapshot();
  });

  it("Copies to clipboard", () => {
    global.document.queryCommandSupported = () => {
      return {
        copy: () => {
          true;
        }
      };
    };
    global.document.execCommand = () => {
      return {
        copy: () => {
          true;
        }
      };
    };
    const tree = mount(<ShareTool shareUrl={testjson.embed_url} />);
    tree.find("button.ShareTool--open-button").simulate("click");
    tree.find("button.ShareTool--copy-button").simulate("click");
    expect(tree).toMatchSnapshot();
    expect(tree.state("copySuccess")).toEqual("Copied!");
  });

  it("Can be closed", () => {
    const tree = mount(<ShareTool shareUrl={testjson.embed_url} />);
    tree.setState({ isShowing: true });
    tree.find("div.ShareTool--modal").simulate("click");
    expect(tree).toMatchSnapshot();
    expect(tree.state("copySuccess")).toEqual("");
  });
});
