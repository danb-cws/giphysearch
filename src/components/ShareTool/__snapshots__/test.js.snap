// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareTool tests Renders and matches snapshot 1`] = `
<button
  className="ShareTool--open-button"
  onClick={[Function]}
/>
`;
